src/zfs-utils/zfs-utils.initcpio.hook

```diff
@@ -44,14 +44,33 @@ zfs_decrypt_fs() {
     # Export encryption root to be used by other hooks (SSH)
     echo "${encryptionroot}" > /.encryptionroot
 
-    # If key location is a file, determine if it can by overridden by prompt
     prompt_override=""
-    if keylocation="$(zfs get -H -o value keylocation "${dataset}")"; then
+    if keylocation="$(zfs get -H -o value keylocation "${encryptionroot}")"; then
```

Comment:
Get keylocation value from encryption root because dataset keylocation might be `none`.